### PR TITLE
🎓 Objects finish introduction topic

### PR DESCRIPTION
**Pull request by jameshughes89**
### What

Finish the introduction topic of Objects. 
- Explain "constructors"
- Explain attributes
- Write methods and use them
- Creating and using instances

### Additional Notes

- I opted not to get into how `__init__` is not actually a constructor as this always creates some confusion and headaches. I will just mention this briefly in lecture
- I opted not to explain how it doesn't need to be `self`, because again, this leads to annoyingness for the class. 
- There is a note on `self` that I am not sure is clear and I worry it makes things _less_ clear. I will note this below.

### CODE REVIEW

**Comment by jameshughes89** [comment]
The diff yucky

**Comment by jameshughes89** [comment]
> The diff yucky

On that --- the content is all new. The deleted stuff was not altered, it was just yeeted. No real need to compare.

File: site/topics/objects/introduction.rst [hunk 1]
```diff
@@ -82,255 +82,207 @@ Starting the Class
...
-            It's going to have a few attributes and some simple functions.
-            '''
+* In the above example, we see the use of the special function called ``__init__``, which describes how to initialize an instance of the class
 
-            def __init__(self, first_name='John', last_name='Doe', student_num='000000000', current_avg=0):
-                self.first_name = first_name 
-                self.last_name = last_name
-                self.student_num = student_num
-                self.current_avg = current_avg
+    * The ``__init__`` method is called the *instantiation* method, but is commonly referred to as the *constructor*
```
Comment (twentylemon) [inline site/topics/objects/introduction.rst:118]: `__init__` is the initializer/constructor. `__new__` actually instantiates the class.

File: site/topics/objects/introduction.rst [hunk 2]
```diff
@@ -82,255 +82,207 @@ Starting the Class
...
 
-    1. Add this function to your code. 
+.. note::
 
-    2. Now in your other script call the ``__repr__() `` method on the ``a_student`` object.
+    Consider the line ``circle_a.radius = 1``. The variable ``circle_a`` is a reference to some ``Circle`` object, and
+    from that object I am assigning the value of ``1`` to that object's ``radius`` attribute. On this line of code, the
+    ``self`` reference variable within the ``Circle`` class is an alias for the object referenced by ``circle_b``.
+    However, on the following line, ``circle_b.radius = 5``, the ``self`` within the ``Circle`` class will be an alias
+    for the object referenced to by ``circle_b``.
```
[the remark this answers is not in the document]
Comment (twentylemon) [reply site/topics/objects/introduction.rst:152]: I don't think this quite makes sense yet. Wait for a method invocation. There's no `self` in this context yet. Imo, just leave it as incantation until then.

Once you have `circle_a.diameter()`, I think it'll be a lot clearer to say that inside that method, `self` is `circle_a`.

File: site/topics/objects/introduction.rst [hunk 3]
```diff
@@ -115,13 +120,13 @@ Constructor/Instantiation and Attributes
 
 * In the above example, we see the use of the special function called ``__init__``, which describes how to initialize an instance of the class
 
-    * The ``__init__`` method is called the *instantiation* method, but is commonly referred to as the *constructor*
+    * The ``__init__`` method is called the *constructor*, or the *instantiation* method
```
Comment (twentylemon) [inline site/topics/objects/introduction.rst:123]: Again, it's not _instantiation_. That doesn't even have _init_ in it.

File: site/topics/objects/introduction.rst [hunk 4]
```diff
@@ -82,255 +82,218 @@ Starting the Class
...
+* When calling ``circle_a.area()``, I am asking the ``Circle`` object referenced by ``circle_a`` to calculate and return its area
+* Similarly, when calling ``circle_b.circumference()``, I am asking the instance ``circle_b`` to calculate and return its circumference
+
+.. note::
+
+    Consider ``circle_a.area()``. The variable ``circle_a`` is a reference to some ``Circle`` object, and I am asking
+    for that instance to calculate and return its ``area``. Here, ``circle_a`` and the ``self`` variable from within the
+    ``Circle`` class are referencing the same ``Circle`` object. When looking at the ``area`` method's code, it makes
+    use of its attribute ``radius``, which is accessed via a reference variable to the instance of the ``Circle`` object
+    the method was invoked on --- ``self``.
```
[the remark this answers is not in the document]
Comment (twentylemon) [reply site/topics/objects/introduction.rst:270]: Think it's clear here. Hammering it home in `__eq__` would be a good thing though.

**Review by twentylemon — APPROVED** [review]

File: site/topics/objects/introduction.rst (hunk 3 above)
Comment (jameshughes89) [reply site/topics/objects/introduction.rst:123]: I always assumed it was initialization, but thought I saw the other today. Upon further investigation, no, it's just straight up initialization.

Comment (jameshughes89) [reply site/topics/objects/introduction.rst:123]: I even thought to myself _init is a dumb choice for instantiation_

**Pull request merged**